drivers/nutdrv_qx.c

```diff
@@ -1813,18 +1814,21 @@ static int	armac_command(const char *cmd, char *buf, size_t buflen)
 	memset(buf, 0, buflen);
 
 	bufpos = 0;
-	while (bufpos + 6 < buflen) {
+	while (bufpos + READ_SIZE < buflen) {
 		size_t bytes_available;
 
 		/* Read data in 6-byte chunks */
 		ret = usb_interrupt_read(udev,
 			0x81,
-			(usb_ctrl_charbuf)tmpbuf, 6, 1000);
+			(usb_ctrl_charbuf)tmpbuf, READ_SIZE, 1000);
 
 		/* Any errors here mean that we are unable to read a reply
 		 * (which will happen after successfully writing a command
 		 * to the UPS) */
-		if (ret != 6) {
+		if (ret != READ_SIZE) {
```

Comment:
Ugh,

````
comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   if (ret != READ_SIZE) {
````

need to cast (probably `(int)READ_SIZE` to become signed in this case).

Reply:
I should have remote access to my UPS tomorrow. I'll try to check it then.

Reply:
Sounds great, thanks for the update!

Reply:
I've finally changed READ_SIZE to int, because the argument in `usb_read_interrupt` is int as well.

My family. Well - that was funny. My sister enabled the UPS, but it was ON BATTERY. So I had a limited time window before it discharged to test the changes. It made me make some changes though. Original code didn't work well (ended reading on \r, then some additional messages were left in buffer and next commands failed).

I added a buffer cleanup at the beginning which seem to work nice and handle more cases. That made it work pretty stable on my UPS as far as I can tell after the tests.